include/dt-bindings/interrupt-controller/intel-ioapic.h

```diff
@@ -10,9 +10,22 @@
 #define	IRQ_TYPE_EDGE			0x00000000
 #define	IRQ_TYPE_LOW			0x00002000
 #define	IRQ_TYPE_HIGH			0x00000000
-#define IRQ_TYPE_EDGE_RISING		(IRQ_TYPE_EDGE | IRQ_TYPE_HIGH)
-#define IRQ_TYPE_EDGE_FALLING		(IRQ_TYPE_EDGE | IRQ_TYPE_LOW)
-#define IRQ_TYPE_LEVEL_HIGH		(IRQ_TYPE_LEVEL | IRQ_TYPE_HIGH)
-#define IRQ_TYPE_LEVEL_LOW		(IRQ_TYPE_LEVEL | IRQ_TYPE_LOW)
+#define	IRQ_DELIVERY_LOWEST		0x00000100
+#define	IRQ_DELIVERY_FIXED		0x00000000
```

Comment:
Why do you need to encode this explicitly?

Reply:
Take an UART interrupt as example, all the way from the generated code (e.g. uart_ns16650_port_0.h) to the ioapic driver:

irq_connect_dynamic() -> arch_irq_connect_dynamic() ->z_irq_controller_irq_config() ->z_ioapic_irq_set().

The only way to config different interrupt with different delivery mode (fixed vs lowest priority) is through the irq_flags that is generated from the .dts.

Thus, the easiest approach available is to change these macros as I did in this patch. Other alternatives that I can think of:

1) Don't change the name of IRQ_TYPE_xxx macros, but silently change it's content:
| #define IRQ_TYPE_LEVEL_LOW	(IRQ_TYPE_LEVEL | IRQ_TYPE_LOW|IRQ_DELIVERY_LOWEST) |

2) Add one more cell to the interrupts property. But this seems a big change. For example:
| interrupts = <4 IRQ_TYPE_EDGE_RISING IRQ_DELIVERY_LOWEST_PRIO 3>|

Reply:
Take an UART interrupt as example, all the way from the generated code (e.g. uart_ns16650_port_0.h) to the ioapic driver:

irq_connect_dynamic() -> arch_irq_connect_dynamic() ->z_irq_controller_irq_config() ->z_ioapic_irq_set().

The only way to config different interrupt with different delivery mode (fixed vs lowest priority) is through the irq_flags that is generated from the .dts.

Thus, the easiest approach available is to change these macros as I did in this patch. Other alternatives that I can think of:

Don't change the name of IRQ_TYPE_xxx macros, but silently change it's content:
 #define IRQ_TYPE_LEVEL_LOW (IRQ_TYPE_LEVEL | IRQ_TYPE_LOW|IRQ_DELIVERY_LOWEST) 

Add one more cell to the interrupts property. But this seems a big change. For example:
 interrupts = <4 IRQ_TYPE_EDGE_RISING IRQ_DELIVERY_LOWEST_PRIO 3>